src/commands/manual-exec.ts

```diff
@@ -107,21 +113,46 @@ export async function manualExec(
 async function doManuallyExecuteSolana(
   payer: Keypair,
   destination: AnchorProvider,
-  transaction: VersionedTransaction,
+  transactions: VersionedTransaction[],
   cluster: string,
 ) {
-  transaction.sign([payer])
-
-  const signature = await destination.connection.sendTransaction(transaction)
-  const latestBlockhash = await destination.connection.getLatestBlockhash()
-  await destination.connection.confirmTransaction(
-    {
-      signature,
-      blockhash: latestBlockhash.blockhash,
-      lastValidBlockHeight: latestBlockhash.lastValidBlockHeight,
-    },
-    'confirmed',
-  )
+  async function waitForFinalization(
+    connection: Connection,
+    signature: string,
+    intervalMs = 500,
+    maxAttempts = 100,
+  ): Promise<void> {
+    for (let attempt = 0; attempt < maxAttempts; attempt++) {
+      const status = await connection.getSignatureStatuses([signature])
+      const info = status.value[0]
+
+      if (info?.confirmationStatus === 'finalized') {
+        return
+      }
+      await new Promise((res) => setTimeout(res, intervalMs))
+    }
+
+    throw new Error(`Transaction ${signature} not finalized after timeout`)
+  }
```

Comment:
I think it makes sense to move this function out to the main level in this file, so it can be re-used later if desired. It is pretty generic, I think it makes sense to not have it nested inside the `doManuallyExecute`